crowbar_framework/app/models/backup.rb

```diff
@@ -0,0 +1,165 @@
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "find"
+
+class Backup < ActiveRecord::Base
+  attr_accessor :file
+
+  before_validation :save_or_create_archive, on: :create
+  after_destroy :delete_archive
+
+  validates :name,
+    presence: true,
+    uniqueness: true,
+    format: {
+      with: /\A[a-zA-Z0-9\-_]+\z/,
+      message: "allows only letters and numbers"
+    }
+  validates :version,
+    presence: true
+  validates :size,
+    presence: true
+
+  validate :validate_file_extension, :validate_version, :validate_hostname
+
+  def exist?
```

Comment:
IMHO this method is obsolete as it doesnt check anything on the model itself. You should directly call exist on the path.